file-avail-s3.md

```diff
@@ -0,0 +1,235 @@
+- **Remove data**: We expect that users remove data from S3 buckets AS
+SOON AS IT IS NO LONGER NEEDED FOR ACTIVELY-RUNNING JOBS.
+- **Monitor usage and quota**: Your account has both a size and
+number of files quota that applies across all buckets owned by your
+account. Quota changes can be requested by emailing chtc@cs.wisc.edu.
+
+CHTC staff reserve the right to remove S3 buckets or revoke bucket
+creation permission at any time.
+
+## D. Data Access Within Jobs
+
+Data in a CHTC S3 bucket is available anywhere (including OSG) as long
+as the version of HTCondor running on the execute machines support
+jobs that use S3 URLs.
+
+Data in CHTC S3 buckets are owned by the user (or a set of users), and
+only the user's (or users') own jobs can access these files.
+
+# 2. Staging Large Data
+
+In order to stage data in an S3 bucket for use on CHTC's HTC system:
+
+- **Get S3 bucket creation access**: Bucket creation access is granted by request.
+- **Create an S3 bucket**: Create a bucket that will contain the data for your project.
+- **Reduce file counts**: Combine and compress files that are used together.
+- **Use the transfer server**: Upload your data to your bucket via our dedicated file transfer server.
+- **Remove files after jobs complete**: Data in S3 buckets are quota controlled and not backed up.
+
+## A. Get S3 Bucket Creation Access
+
+CHTC S3 bucket creation access is granted by request. If you think you need
+to create S3 buckets, email CHTC's Research Computing Facilitators (chtc@cs.wisc.edu).
+
+## B. Create an S3 Bucket
+
+Buckets can be created on a CHTC submit server or the CHTC transfer server
+using the `mc` command:
+
+``` {: .term}
+$ mc mb chtc/my-bucket-name
+```
+
+Each bucket in CHTC must have a unique name, so be descriptive!
+
+## C. Reduce File Counts
+
+Data placed in S3 buckets should be stored in as few files as possible
+(ideally, one file per job). Similarly, large output should first be
+written to the job working directory then compressed in to a single
+file before being transferred back to an S3 bucket at the end of the job.
+
+To prepare job-specific data that is large enough
+and exists as multiple files or directories (or a directory of multiple
+files), first create a compressed tar package before placing the file in
+an S3 bucket (either before submitting jobs, or within jobs before
+transferring output to). For example:
+
+``` {: .term}
+$ tar -czvf job_package.tar.gz file_or_dir
+```
+
+## D. Use the Transfer Server
+
+Movement of large data into/out of S3 buckets before and after jobs
+should be performed via CHTC's transfer server, as below, and
+**not via a CHTC submit server.** After obtaining an account on the
+transfer server and creating an S3 bucket, copy relevant files directly into your
+home directory from your own computer:
+
+- Example `scp` command on your own Linux or Mac computer:
+``` {.term}
+$ scp large-input.file username@transfer.chtc.wisc.edu:/home/username/
```

Comment:
what if the `/home` quotas are too small to do this? I think we set them extra low on the transfer server because we don't want people storing data there...

Reply:
I don't know yet what the recommended solution is for people to upload directly to S3 from their home (or lab or office etc) computers yet, but that should definitely be listed as an option here.

Reply:
Agree.

Reply:
Ok, we'll add this when we figure out a recommendation for outside-CHTC transfers (probably Globus?).